.github/workflows/publish_precompiled.yml

```diff
@@ -0,0 +1,86 @@
+name: Publish Precompiled
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+  S3_BUCKET: s3.hex.pm
+
+
+jobs:
+  publish_precompiled:
+    strategy:
+      fail-fast: true
+      # Prevent each job edit builds.txt concurrently.
+      max-parallel: 1
+      matrix:
+        include:
+          - otp: 24
+            otp_version: 24.3
+          - otp: 25
+            otp_version: 25.0
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 50
+
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp_version }}
+          version-type: strict
+
+      - name: Build Elixir Release
+        run: |
+          make Precompiled.zip
+          mv Precompiled.zip main-otp-${{ matrix.otp }}.zip
+
+      - name: Update builds.txt
+        run: |
+          date=$(date -u '+%Y-%m-%dT%H:%M:%SZ')
+          build_sha256=$(sha256sum main-otp-${{ matrix.otp }}.zip | cut -d ' ' -f 1)
+
+          aws s3 cp s3://${{ env.S3_BUCKET }}/builds/elixir/builds.txt builds.txt
+          sed -i '/^main-otp-${{ matrix.otp }} /d' builds.txt
+          echo -e -n "main-otp-${{ matrix.otp }} ${{ github.sha }} ${date} ${build_sha256}\n$(cat builds.txt)" > builds.txt
+          sort -u -k1,1 -o builds.txt builds.txt
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.HEXPM_ELIXIR_BUILDS_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.HEXPM_ELIXIR_BUILDS_AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-1
+
+      - name: Upload to S3
+        run: |
+          for file in main-otp-${{ matrix.otp }}.zip builds.txt
+          do
+            aws s3 cp "${file}" s3://${{ env.S3_BUCKET }}/builds/elixir/ \
+              --cache-control 'public,max-age=3600' \
+              --metadata '{"surrogate-key":"builds","surrogate-control":"public,max-age=604800"}'
+          done
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.HEXPM_ELIXIR_BUILDS_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.HEXPM_ELIXIR_BUILDS_AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-1
+
+  fastly_purge:
+    needs: publish_precompiled
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Purge Fastly cache
+        run: |
+          curl \
+            --fail \
+            -X POST \
+            -H "Fastly-Key: ${BOB_FASTLY_KEY}" \
+            -H "Accept: application/json" \
+            -H "Content-Length: 0" \
+            "https://api.fastly.com/service/${BOB_FASTLY_SERVICE}/purge/${BOB_FASTLY_KEY}"
+        env:
+          BOB_FASTLY_KEY: ${{ secrets.BOB_FASTLY_KEY }}
+          BOB_FASTLY_SERVICE: changeme
```

Comment:
I'm not sure what service name we should use or just use `secrets` instead?

Reply:
Lets use secret here as well.